datalad/api.py

```diff
@@ -25,7 +25,26 @@ def _load_plugins():
         globals()[camel.sub('\\1_\\2', pi.__name__).lower()] = pi.__call__
 
 
+def _generate_modules_api():
+    """Auto detect all available modules and generate an API from them
+    """
+    from importlib import import_module
+    from pkg_resources import iter_entry_points
+    from .interface.base import get_api_name
+
+    for entry_point in iter_entry_points('datalad.modules'):
```

Comment:
why should it be a separate namespace/special?  i.e. shouldn't we then define our internal functionality as modules?  This mechanism is using "Interface", so may be this one should just be "datalad.interface"?

Reply:
Sure, but maybe one step after the other. Who knows how/why this breaks. Also interface is already heavily overloaded. Primary objective here is to allows for 3rd-party extensions, not for RF'ing datalad itself. If we find this to be working fine, we can adopt it in general. but please remember that _you_ remove the use of setuptools entrypoints for the main cmdline entrypoint, because it was sooo slow.

Reply:
yeap, I remember that I did it.